dot/network/gossip.go

```diff
@@ -24,13 +25,17 @@ func newGossip() *gossip {
 }
 
 // hasSeen broadcasts messages that have not been seen
-func (g *gossip) hasSeen(msg NotificationsMessage) bool {
+func (g *gossip) hasSeen(msg NotificationsMessage) (bool, error) {
```

Comment:
nit named returns would be nice

```suggestion
func (g *gossip) hasSeen(msg NotificationsMessage) (seen bool, err error) {
```